RustSyncManager crashes with persistedState=null

A `null` persisted state is our "first sync" scenario.

```
                         E  Caused by: java.lang.NullPointerException
                         E      at mozilla.appservices.syncmanager.MsgTypes$SyncParams.setPersistedState(MsgTypes.java:776)
                         E      at mozilla.appservices.syncmanager.MsgTypes$SyncParams.access$1100(MsgTypes.java:499)
                         E      at mozilla.appservices.syncmanager.MsgTypes$SyncParams$Builder.setPersistedState(MsgTypes.java:1605)
                         E      at mozilla.appservices.syncmanager.SyncParams.toProtobuf$syncmanager_release(SyncParams.kt:144)
                         E      at mozilla.appservices.syncmanager.SyncManager.sync(SyncManager.kt:48)
                         E      at mozilla.components.service.fxa.sync.WorkManagerSyncWorker.doWork(WorkManagerSyncManager.kt:328)
                         E      at androidx.work.CoroutineWorker$startWork$1.invokeSuspend(CoroutineWorker.kt:66)
                         E      at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
                         E      at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:241)
                         E      at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:594)
                         E      at kotlinx.coroutines.scheduling.CoroutineScheduler.access$runSafely(CoroutineScheduler.kt:60)
                         E      at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:740)
```

┆Issue is synchronized with this [Jira Story](https://jira.allizom.org/browse/SYNC-447)
┆Story Points: 1
┆Sprint: SYNC - 2019-10-11
┆Epic: [Q3 1.1] Sync Manager / CWTS

--- a/components/sync_manager/android/src/main/java/mozilla/appservices/syncmanager/SyncParams.kt
+++ b/components/sync_manager/android/src/main/java/mozilla/appservices/syncmanager/SyncParams.kt
@@ -140,8 +140,7 @@ data class SyncParams(
         builder.acctSyncKey = this.authInfo.syncKey
         builder.acctKeyId = this.authInfo.kid
         builder.acctTokenserverUrl = this.authInfo.tokenserverURL
-
-        builder.persistedState = this.persistedState
+        this.persistedState?.let { builder.persistedState = it }
 
         builder.fxaDeviceId = this.deviceSettings.fxaDeviceId
         builder.deviceName = this.deviceSettings.name

